src/types/chat_member.rs

```diff
@@ -154,227 +244,334 @@ impl ChatMemberKind {
         }
     }
 
+    /// Returns `true`, if the user's presence in the chat is hidden.
+    ///
+    /// I.e. returns `true` if the user is the creator of the chat or an
+    /// administrator in the chat and has [`can_manage_chat`] privilege.
+    /// Returns `false` otherwise.
+    ///
+    /// [`can_manage_chat`]: Administrator::can_manage_chat
+    pub fn is_anonymous(&self) -> bool {
+        match self {
+            Self::Creator(Creator { is_anonymous, .. })
+            | Self::Administrator(Administrator { is_anonymous, .. }) => *is_anonymous,
+            Self::Member | Self::Restricted(_) | Self::Left | Self::Kicked(_) => false,
+        }
+    }
+
     /// Getter for [`Restricted::until_date`] and [`Kicked::until_date`] fields.
-    pub fn until_date(&self) -> Option<i32> {
+    pub fn until_date(&self) -> Option<i64> {
         match &self {
             Self::Creator(_) | Self::Administrator(_) | Self::Member | Self::Left => None,
             Self::Restricted(Restricted { until_date, .. })
             | Self::Kicked(Kicked { until_date, .. }) => Some(*until_date),
         }
     }
+}
 
-    /// Getter for [`Administrator::can_be_edited`] field.
-    pub fn can_be_edited(&self) -> Option<bool> {
-        match &self {
-            Self::Administrator(Administrator { can_be_edited, .. }) => Some(*can_be_edited),
+/// Methods to check admin privileges.
+impl ChatMemberKind {
+    /// Returns `true` if the user is an administrator in the given chat and the
+    /// bot is allowed to edit administrator privileges of that user.
+    pub fn can_be_edited(&self) -> bool {
+        match self {
+            Self::Administrator(Administrator { can_be_edited, .. }) => *can_be_edited,
+            // Creator can't ever be edited by any bot.
             Self::Creator(_)
             | Self::Member
             | Self::Restricted(_)
             | Self::Left
-            | Self::Kicked(_) => None,
+            | Self::Kicked(_) => false,
         }
     }
 
-    /// Getter for [`Administrator::can_manage_chat`] field.
-    pub fn can_manage_chat(&self) -> Option<bool> {
-        match &self {
+    /// Returns `true`, if the user can access the chat event log, chat
```

Comment:
```suggestion
    /// Returns `true` if the user can access the chat event log, chat
```

The `<X>, if` form is generally acceptable but in this case is less smooth.